pkg/kubelet/cm/dra/manager.go

```diff
@@ -379,7 +382,7 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 			if reqClaim == nil {
 				return fmt.Errorf("NodeUnprepareResources returned result for unknown claim UID %s", claimUID)
 			}
-			if result.Error != "" {
+			if result != nil && result.Error != "" {
```

Comment:
I just remembered that there is https://github.com/kubernetes/kubernetes/blob/f4e246bc93ffb68b33ed67c7896c379efa4207e7/staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha3/api.pb.go#L338

Using that would make the code a bit simpler.